File: Backend/src/main/java/com/github/francisfire/anavis/models/Prenotation.java
package com.github.francisfire.anavis.models;

import java.util.Date;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import org.springframework.data.annotation.Id;

@Getter
@Setter
@NoArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
abstract class Prenotation {

	@EqualsAndHashCode.Include
	@Id
	private String id;
	private String officeMail;
	private String donorMail;
	private Date hour;

	public Prenotation(String id, String officeMail, String donorMail, Date hour) {
		this.id = id;
		this.officeMail = officeMail;
		this.donorMail = donorMail;
		this.hour = hour;
	}

}

File: Frontend/lib/models/prenotation.dart
abstract class Prenotation {
  String _id;
  String _officeMail;
  String _donorMail;
  String _hour;
  Prenotation.withParams(
      String id, String officeMail, String donorMail, String hour) {
    this._id = id;
    this._officeMail = officeMail;
    this._donorMail = donorMail;
    this._hour = hour;
  }

  Prenotation() {}
  String getId() {
    return _id;
  }

  String getOfficeMail() {
    return _officeMail;
  }

  String getDonorMail() {
    return _donorMail;
  }

  String getHour() {
    return _hour;
  }
}
